src/profiler.cpp

```diff
@@ -614,6 +618,11 @@ void Profiler::fillFrameTypes(ASGCT_CallFrame* frames, int num_frames, NMethod*
 u64 Profiler::recordSample(void* ucontext, u64 counter, EventType event_type, Event* event) {
     atomicInc(_total_samples);
 
+    if (_last_epoch != _epoch) {
```

Comment:
I'm pretty sure this should be read with some level of synchronization, but you can leave it as it is for now.